docs/docs/recipes/create-your-connector/connector-implementation-guide.mdx

```diff
@@ -3,44 +3,261 @@ sidebar_label: Guide to implement connectors
-_SMS/Email Connector_ doc keyword: integrate SMS/Email sender API.
 :::
 
-## Implementation and critical information composition
+In order to accomplish the flow, we need to have following three methods:
+
+### `getAuthorizationUri`
+
+`getAuthorizationUri` directs end-users to the page that need their authentication.
+
+Listed parameters are required:
+
+- `authorizationEndpoint` can be found in GitHub OAuth doc site, which is the page where end-user should go for authentication
+- `config`, which includes `clientId` and `clientSecret` in GitHub scenario
+- `state`, a random string to proof CSRF
+- `scope` to determine specific user information
+- `redirectUri` of landing page after end-user's successful authentication.
+
+```typescript
+const getAuthorizationUri = async ({ state, redirectUri }) => {
+  const queryParameters = new URLSearchParams({
+    client_id: config.clientId, // `config` is your GitHub application credential
+    redirect_uri: redirectUri,
+    state,
+    scope,
+  });
+
+  return `${authorizationEndpoint}?${queryParameters.toString()}`;
+};
+```
+
+### `getAccessToken`
+
+`getAccessToken` gets access token with authorization code issued after end-users successful authentication.
+
+Besides `config` we mentioned in previous `getAuthorizationUri` method, we also want to get:
+
+- authorization `code` from parameters brought to redirect landing page
+- `accessTokenEndpoint`, which is the endpoint to get access token with authorization code
+
+```typescript
+const getAccessToken = async (config: GithubConfig, code: string) => {
+  const { clientId: client_id, clientSecret: client_secret } = config;
+
+  const httpResponse = await got.post({
+    url: accessTokenEndpoint,
+    json: {
+      client_id,
+      client_secret,
+      code,
+    },
+    timeout: defaultTimeout,
+  });
+
+  const result = accessTokenResponseGuard.safeParse(qs.parse(httpResponse.body));
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error);
+  }
+
+  const { access_token: accessToken } = result.data;
+
+  assert(accessToken, new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid));
+
+  return { accessToken };
+};
+```
+
+### `getUserInfo`
+
+`getUserInfo` fetches user information with access token got in previous step.
+
+`userInfoEndpoint` is the endpoint which is used to get user information.
+
+You may check official documents to find specific user information that can be accessed at user info endpoint and corresponding scope.
+
+`id` assigned by the connector's service provider (in this case, GitHub) is required; other information is optional. If you can get `email` or `phone` from the user profile, MAKE SURE that they are "verified" since they will be used to receive passcodes in passwordless flows. You also need to align the returning keys with the fields name in [Logto user profile](../../references/users/README.md#basic-data).
```

Comment:
The reason why they should be verified is not accurate, it's hard to describe, may be delete the "since"?

Reply:
Maybe they don't necessarily have to be "verified" any more, since we can use the dedicated "email_verified" and "phone_verified" properties?

Reply:
we added "email/phone_verified" to the return value of SDK to align with the protocol, and these two values will always be `true` per our current system/DB design. Will revisit this after "profile fulfilling" ends to see if unverified email/phone should be supported.
TL;DR: keep returning verified email/phone from users' profiles. cc: @charIeszhao